FizzBuzz/src/main/java/rocks/kata/FizzBuzzDivisibilityRules.java

```diff
@@ -0,0 +1,65 @@
+package rocks.kata;
+
+import java.util.Map;
+import static rocks.kata.DivisibilityStatus.*;
+
+public class FizzBuzzDivisibilityRules implements FizzBuzz {
+
+    private Map<DivisibilityStatus, String> fizzBuzzTranslation = Map.of(
+            DIVISIBLE_BY_3, "Fizz",
+            DIVISIBLE_BY_5, "Buzz",
+            DIVISIBLE_BY_15, "FizzBuzz");
+
+    @Override
+    public String returnFizzBuzzOrNumber(int number) {
+        DivisibilityStatus checkDivisibilityStatus = checkDivisibility(number);
+
+        return this.fizzBuzzTranslation.getOrDefault(checkDivisibilityStatus, Integer.toString(number));
+    }
+
+    private DivisibilityStatus checkDivisibility(int number) {
+        if (isDivisibleBy15(number)) {
+            return DIVISIBLE_BY_15;
+        }
+        if (isDivisibleBy3(number)) {
+            return DIVISIBLE_BY_3;
+        }
+        if (isDivisibleBy5(number)) {
+            return DIVISIBLE_BY_5;
+        }
+        return NO_MATCHING_DIVISIBILITY;
+
+    }
+
+    private boolean isDivisibleBy3(int number) {
+        int helper = number;
+        do {
+            helper = calculateSumOfDigits(helper);
+        } while (helper >= 10);
```

Comment:
Przydałby się komentarz do tej pętli, bo nie jest na pierwszy rzut oka zrozumiały ten algorytm, a to doskonały powód, żeby dodać komentarz.